packages/big-design/src/components/Worksheet/editors/ModalEditor/ModalEditor.tsx

```diff
@@ -1,32 +1,46 @@
-import React from 'react';
+import React, { createRef, useCallback, useEffect, useMemo } from 'react';
 
 import { typedMemo } from '../../../../utils';
-import { Flex, FlexItem } from '../../../Flex';
+import { Flex } from '../../../Flex';
 import { Small } from '../../../Typography';
 import { useStore } from '../../hooks';
 import { Cell, WorksheetItem, WorksheetModalColumn } from '../../types';
 
-import { StyledButton } from './styled';
+import { StyledButton, StyledFlexItem } from './styled';
 
 export interface ModalEditorProps<Item> {
   cell: Cell<Item>;
+  isEditing: boolean;
   formatting?: WorksheetModalColumn<Item>['formatting'];
 }
 
-const InternalModalEditor = <T extends WorksheetItem>({ cell, formatting }: ModalEditorProps<T>) => {
+const InternalModalEditor = <T extends WorksheetItem>({ cell, formatting, isEditing }: ModalEditorProps<T>) => {
   const setOpenModal = useStore((state) => state.setOpenModal);
+  const setEditingCell = useStore((state) => state.setEditingCell);
   const { hash, value } = cell;
 
-  const handleClick = () => {
-    setOpenModal(hash);
-  };
+  const buttonRef = createRef<HTMLButtonElement>();
+
+  useEffect(() => {
+    if (isEditing) {
+      setOpenModal(hash);
+    }
+  }, [hash, isEditing, setOpenModal]);
+
+  const handleClick = useCallback(() => {
```

Comment:
🍹 We could give this a more descriptive name (`handleEditCell` or something), but it's up to you 👍

Reply:
Still prefer `handleClick` since it handles the click of the element.